cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -621,58 +621,61 @@ public LoadBalancer getLoadBalancer(String resourceGroupName, String loadBalance
                 return List.of();
 
         }
     }
 
-    private List<String> getLoadBalancerIps(String resourceGroupName, String loadBalancerName) {
+    private List<AzureLoadBalancerFrontend> getLoadBalancerFrontends(String resourceGroupName, String loadBalancerName) {
         List<String> idsAssociatedWithLoadBalancerPublicIps = getLoadBalancer(resourceGroupName, loadBalancerName).publicIpAddressIds();
 
-        List<String> loadBalancerIps = getPublicIpAddresses(resourceGroupName)
+        List<AzureLoadBalancerFrontend> frontends = getPublicIpAddresses(resourceGroupName)
                 .getAll()
                 .stream()
                 .filter(ipAddress -> idsAssociatedWithLoadBalancerPublicIps.contains(ipAddress.id()))
-                .map(PublicIpAddress::ipAddress)
-                .sorted()
+                .map(ipAddress ->
+                        new AzureLoadBalancerFrontend(ipAddress.getAssignedLoadBalancerFrontend().name(), ipAddress.ipAddress(), LoadBalancerType.PUBLIC))
                 .collect(Collectors.toList());
 
-        LOGGER.info("IPs for load balancer {} retrieved: {}", loadBalancerName, loadBalancerIps);
-        return loadBalancerIps;
+        LOGGER.info("Frontends for public load balancer {} retrieved: {}", loadBalancerName, frontends);
+        return frontends;
     }
 
-    private List<String> getLoadBalancerPrivateIps(String resourceGroupName, String loadBalancerName) {
-        // The keys in this map are the names of the frontend load balancers. We don't use them however.
-        Map<String, LoadBalancerFrontend> frontends = getLoadBalancer(resourceGroupName, loadBalancerName).frontends();
+    private List<AzureLoadBalancerFrontend> getLoadBalancerPrivateFrontends(String resourceGroupName, String loadBalancerName) {
+        // The keys in this map are the names of the frontend load balancers.
+        Map<String, LoadBalancerFrontend> providerFrontends = getLoadBalancer(resourceGroupName, loadBalancerName).frontends();
 
-        List<String> loadbalancerPrivateIps = frontends.values().stream()
-                .map(LoadBalancerFrontend::innerModel)
-                .map(FrontendIpConfigurationInner::privateIpAddress)
-                .filter(Objects::nonNull)
-                .sorted()
+        List<AzureLoadBalancerFrontend> frontends = providerFrontends.entrySet().stream()
+                .filter(fe -> Objects.nonNull(fe.getValue().innerModel())
+                        && Objects.nonNull(fe.getValue().innerModel().privateIpAddress())
+                        && !fe.getValue().isPublic())
+                // The LoadBalancerType pseudo-type is matched based on the frontend name. It is unfortunate, I know.
+                .map(fe ->
+                        new AzureLoadBalancerFrontend(fe.getKey(), fe.getValue().innerModel().privateIpAddress(),
+                                fe.getKey().endsWith("gateway") ? LoadBalancerType.GATEWAY_PRIVATE : LoadBalancerType.PRIVATE))
```

Comment:
again maybe this should the method inside the class

Reply:
this is also part of the Azure SDK